cadctap/cadctap/tests/test_core.py

```diff
@@ -508,6 +509,60 @@ def test_help(self):
                 main_app()
             self.assertEqual(usage, stdout_mock.getvalue())
 
+    @patch('sys.exit', Mock(side_effect=[MyExitError, MyExitError, MyExitError,
+                                         MyExitError, MyExitError, MyExitError,
+                                         MyExitError, MyExitError]))
+    def test_exit_on_exception(self):
+        """ Test the exit_on_exception function """
+
+        # test handling of 'Bad Request' server errors
+        with patch('sys.stderr', new_callable=StringIO) as stderr_mock:
+            expected_message = "source error message"
+            orig_ex = RuntimeError("Bad Request")
+            ex = exceptions.HttpException(expected_message, orig_ex)
+            try:
```

Comment:
No need to use try, except - just pass the exception to exit_on_exception

Reply:
That's what I did initially, but python 3.4 did not like it.